src/scheduler/node_info.c

```diff
@@ -3292,6 +3324,85 @@ is_vnode_eligible_chunk(resource_req *specreq, node_info *node,
 
 /**
  * @brief
+ *	Checks if a vnode is eligible for power operations.
+ *  Based on is_provisionable.
+ *
+ * @par Functionality:
+ *	This function checks if a vnode is eligible to be provisioned.
+ *	A vnode is eligible for power operations if it satisfies all of the
+ *	following conditions:-
+ *	(1) Server has power_provisioning True,
+ *	(2) Vnode has power_provisioning True,
+ *	(3) No conflicts with reservations already running on the Vnode
+ *	(4) No conflicts with jobs already running on the Vnode
+ *
+ * @param[in]		node	-	pointer to node_info
+ * @param[in]		resresv	-	pointer to resource_resv
+ * @param[in,out]	err		-	pointer to schd_error
+ *
+ * @return	int
+ * @retval	 NO_PROVISIONING_NEEDED : resresv doesn't request eoe
+ *			or resresv is not a job
+ * @retval	 PROVISIONING_NEEDED : vnode doesn't have current_eoe set
+ *          or it doesn't match job eoe
+ * @retval	 NOT_PROVISIONABLE  : vnode is not provisionable
+ *			(see err for more details)
+ *
+ * @par Side Effects:
+ *	Unknown
+ *
+ * @par MT-safe: No
+ *
+ */
+int
+is_powerok(node_info *node, resource_resv *resresv, schd_error *err)
+{
+	int i;
+	int ret = NO_PROVISIONING_NEEDED;
+
+	if (!resresv->is_job)
```

Comment:
can there be more than 1 reservation (or a job) on the node with different eoe values? If so should we not be checking for that?